pkg/collector/corechecks/network-devices/cisco-sdwan/sdwan.go

```diff
@@ -0,0 +1,211 @@
+}
+
+// Run executes the check
+func (c *CiscoSdwanCheck) Run() error {
+	log.Info("Running Cisco SD-WAN check")
+
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
+	}
+
+	c.metricsSender.SetSender(sender)
+
+	devices, err := c.client.GetDevices()
+	if err != nil {
+		log.Warnf("Error getting devices from Cisco SD-WAN API: %s", err)
+	}
+	vEdgeInterfaces, err := c.client.GetVEdgeInterfaces()
+	if err != nil {
+		log.Warnf("Error getting vEdge interfaces from Cisco SD-WAN API: %s", err)
+	}
+	cEdgeInterfaces, err := c.client.GetCEdgeInterfaces()
+	if err != nil {
+		log.Warnf("Error getting cEdge interfaces from Cisco SD-WAN API: %s", err)
+	}
+	deviceStats, err := c.client.GetDeviceHardwareMetrics()
+	if err != nil {
+		log.Warnf("Error getting device metrics from Cisco SD-WAN API: %s", err)
+	}
+	interfaceStats, err := c.client.GetInterfacesMetrics()
+	if err != nil {
+		log.Warnf("Error getting interface metrics from Cisco SD-WAN API: %s", err)
+	}
+	appRouteStats, err := c.client.GetApplicationAwareRoutingMetrics()
+	if err != nil {
+		log.Warnf("Error getting application-aware routing metrics from Cisco SD-WAN API: %s", err)
+	}
+	controlConnectionsState, err := c.client.GetControlConnectionsState()
+	if err != nil {
+		log.Warnf("Error getting control-connection states from Cisco SD-WAN API: %s", err)
+	}
+	ompPeersState, err := c.client.GetOMPPeersState()
+	if err != nil {
+		log.Warnf("Error getting OMP peer states from Cisco SD-WAN API: %s", err)
+	}
+	bfdSessionsState, err := c.client.GetBFDSessionsState()
+	if err != nil {
+		log.Warnf("Error getting BFD session states from Cisco SD-WAN API: %s", err)
+	}
+	deviceCounters, err := c.client.GetDevicesCounters()
+	if err != nil {
+		log.Warnf("Error getting device counters from Cisco SD-WAN API: %s", err)
+	}
+
+	devicesMetadata := payload.GetDevicesMetadata(c.config.Namespace, devices)
+	deviceTags := payload.GetDevicesTags(c.config.Namespace, devices)
+	uptimes := payload.GetDevicesUptime(devices)
+
+	interfaces := payload.ConvertInterfaces(vEdgeInterfaces, cEdgeInterfaces)
+	interfacesMetadata, interfacesMap := payload.GetInterfacesMetadata(c.config.Namespace, interfaces)
+	ipAddressesMetadata := payload.GetIPAddressesMetadata(c.config.Namespace, interfaces)
+
+	c.metricsSender.SetDeviceTags(deviceTags)
+
+	if !c.config.NoMetadata {
+		c.metricsSender.SendMetadata(devicesMetadata, interfacesMetadata, ipAddressesMetadata)
+	}
+	c.metricsSender.SendDeviceMetrics(deviceStats)
+	c.metricsSender.SendInterfaceMetrics(interfaceStats, interfacesMap)
+	c.metricsSender.SendUptimeMetrics(uptimes)
+	c.metricsSender.SendAppRouteMetrics(appRouteStats)
+	c.metricsSender.SendControlConnectionMetrics(controlConnectionsState)
+	c.metricsSender.SendOMPPeerMetrics(ompPeersState)
+	c.metricsSender.SendBFDSessionMetrics(bfdSessionsState)
+	c.metricsSender.SendDeviceCountersMetrics(deviceCounters)
+
+	// Commit
+	c.metricsSender.Commit()
+	log.Info("Done running Cisco SD-WAN check")
```

Comment:
IIRC this is already logged by the Agent on its own. In general check rarely log anything at the Info level on every check run to avoid noise.

The Agent usually logs some information about the first X check runs, then from time to time.

Reply:
Good catch! removed in cc33032